crates/starknet-server/tests/test_messaging.rs

```diff
@@ -481,6 +481,19 @@ mod test_messaging {
             devnet.post_json("/postman/flush".into(), "".into()).await.expect("flush failed");
         assert_eq!(resp.status(), StatusCode::OK, "Checking status of {resp:?}");
 
+        // Assert traces of L1Handler transaction with custom rpc call,
+        // json_rpc_client.trace_transaction() is not supported
+        let flush_body = get_json_body(resp).await;
+        let l1_handler_tx_trace = &devnet
+            .send_custom_rpc(
+                "starknet_traceTransaction",
+                json!({ "transaction_hash": flush_body.get("generated_l2_transactions").unwrap()[0] }),
+            )
+            .await["result"];
+        assert_eq!(l1_handler_tx_trace["type"], "L1_HANDLER");
+        assert!(l1_handler_tx_trace["function_invocation"].is_null());
+        assert!(l1_handler_tx_trace["state_diff"].is_null());
+
         // Ensure the balance is back to 1 on L2.
         assert_eq!(get_balance(&devnet, sn_l1l2_contract, user_sn).await, [FieldElement::ONE]);
```

Comment:
In any case, this assertion should be done sooner, just after flush response status is checked in L482.

Reply:
ed5595b7bfbce2330aa5cfa29532148f735ddb07